sentry/src/main/java/io/sentry/Transaction.java

```diff
@@ -0,0 +1,134 @@
+package io.sentry;
+
+import io.sentry.protocol.SentryId;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class Transaction extends SentryBaseEvent<TransactionContexts> implements ISpan {
+  /** The transaction name. */
+  private @Nullable String transaction;
+
+  /** The moment in time when span was started. */
+  private final @NotNull Date startTimestamp;
+
+  /** The moment in time when span has ended. */
+  private @Nullable Date timestamp;
+
+  /** A list of spans within this transaction. Can be empty. */
+  private final @NotNull List<Span> spans = new CopyOnWriteArrayList<>();
+
+  /**
+   * A hub this transaction is attached to. Marked as transient to be ignored during JSON
+   * serialization.
+   */
+  private @NotNull final transient IHub hub;
+
+  /** The {@code type} property is required in JSON payload sent to Sentry. */
+  @SuppressWarnings("UnusedVariable")
+  private @NotNull final String type = "transaction";
+
+  /** The {@code platform} property is required in JSON payload sent to Sentry. */
+  @SuppressWarnings("UnusedVariable")
+  private @NotNull final String platform = "java";
```

Comment:
Shouldn't this be on the top-most Event type?
Or: is it even needed for Transactions?

Reply:
I was wrong, it's not required 👍